_posts/2020-08-05-using-open-policy-agent-with-strimzi-and-apache-kafka.md

```diff
@@ -0,0 +1,456 @@
+The ACL rules have to cover all operations which the user is allowed to do.
+You can use prefix or the `*` wildcard to cover multiple resource names with one ACL rule, but that is it.
+It also doesn't support any groups, so even if you have multiple users with the same rights, you need to configure the ACL rules individually for each of them.
+So you often end up with lot of different rules for each user.
+
+The main advantage of the `simple` authorization is that it is ... well ... _simple_.
+The ACL rules for the `simple` authorization are stored inside ZooKeeper.
+When the rules change, the Kafka brokers will be notified about the change and load the new ACLs.
+You do not need to deploy and operate any other application.
+Everything is built into Strimzi, Kafka and ZooKeeper.
+Another advantage is that it can be easily combined with all of our authentication mechanisms.
+It is also supported in the User Operator, so you can configure the ACL rules directly in the `KafkaUser` custom resources.
+So you can easily configure everything in YAML, store it in your Git repository and use GitOps to manage the resources.
+
+## Keycloak Authorization Services
+
+[Keycloak](https://www.keycloak.org/) is an open source identity and access management server.
+Strimzi supports authentication of Kafka clients using OAuth 2.0 access tokens.
+And Keycloak is one of the servers you can use for it (see our older blog post _[Kafka authentication using OAuth 2.0](https://strimzi.io/blog/2019/10/25/kafka-authentication-using-oauth-2.0/)_).
+But you can also use it for authorization thanks to our own `KeycloakRBACAuthorizer`.
+
+The `keycloak` authorization is based on the [Keycloak Authorization Services](https://www.keycloak.org/docs/latest/authorization_services/index.html).
+It can be used only in combination with OAuth 2.0 authentication.
+Security policies and permissions are defined in Keycloak and are used to grant access to resources on Kafka brokers. 
+Users and clients are matched against policies that permit access to perform specific actions on Kafka brokers.
+You can also use the _federation_ feature of Keycloak to map the policies for example against LDAP groups.
+The `KeycloakRBACAuthorizer` will fetch the list of granted permissions from Keycloak and enforce the authorization rules locally to make sure it doesn't negatively impact performance.
+
+The main advantage of Keycloak is that you can have everything managed centrally.
+Not just for Strimzi and Apache Kafka, but for all your applications which support OAuth.
+It can also handle both authentication and authorization.
+So you do not need to have different servers for authentication and authorization and all data are stored in one place.
+
+On the other hand, you will still need to run and manage Keycloak.
+So unless you already use it or at least plan to use some central authentication / authorization server anyway, this will be additional effort you would need to deal with.
```

Comment:
```suggestion
Unless you already use it or at least plan to use some central authentication / authorization server anyway, this will be additional effort you would need to deal with.
```